src/server/src/read-project-folders.ts

```diff
@@ -4,7 +4,14 @@ import { callLimit } from 'promise-call-limit'
 import { ignoredHomedirFolderNames } from './ignored-homedir-folder-names.ts'
 
 const limit = Math.max(availableParallelism() - 1, 1) * 8
-const home = homedir()
+let home: string
+try {
+  home = homedir()
+} catch {
+  // In restricted environments, homedir() might fail.
+  // Fall back to current working directory.
+  home = process.cwd()
+}
```

Comment:
same as mentioned before, I think the better solution is to have no paths in this case, I'd rather change the logic to return early an empty array if we see that `userDefinedProjectPaths.length === 0`